modules/IntegrationTesting/src/scenarioexecution.cpp

```diff
@@ -0,0 +1,154 @@
+		trajectory.push_back(std::make_pair(t.pose.position.x, t.pose.position.y));
+	}
+	return trajectory;
+}
+
+
+/**
+ * @brief Check that the object ends up near the end point of the trajectory.
+ * 
+ */
+void ScenarioExecution::checkTrajectory() {
+	auto trajectory = getTrajectoryPoints();
+	std::pair<double, double> lastPoint;
+	auto isObjectMoving = true;
+	while(isObjectMoving) {
+		atos_interfaces::msg::Monitor monr;
+		monitorSub = this->create_subscription<atos_interfaces::msg::Monitor>("/atos/object_1/object_monitor", 10, std::bind(&ScenarioExecution::placeholderCallback, this, std::placeholders::_1));
+		rclcpp::wait_for_message(monr, monitorSub, this->get_node_base_interface()->get_context(), 50ms);
+		if (monr.velocity.twist.linear.x == 0 && monr.velocity.twist.linear.y == 0) {
+			lastPoint = std::make_pair(monr.pose.pose.position.x, monr.pose.pose.position.y);
+			isObjectMoving = false;
+		}
+	}
+
+	auto distanceX = lastPoint.first - trajectory.back().first;
+	auto distanceY = lastPoint.second - trajectory.back().second;
+	auto distanceThreshold = 1.0;
+	if (abs(distanceX) < distanceThreshold && abs(distanceY) < distanceThreshold) {
+		followedTrajectory = true;
+		RCLCPP_INFO(get_logger(), "Object followed trajectory succesfully");
+	}
+	else {
+		followedTrajectory = false;
+		RCLCPP_ERROR(get_logger(), "Object did not follow trajectory succesfully. Distance from end point of trajectory is x: %f, y: %f", distanceX, distanceY);
+	}
+}
+
+
+/**
+ * @brief Print the result of the integration test.
+ * 
+ */
+void ScenarioExecution::printResult() {
+	std::stringstream ss;
+	auto width = 20;
+	ss << "\nIntegration test - Scenario Excecution results:\n";
+	ss << "State change results table:\n";
+	ss << std::setfill('-') << std::setw(width * 3) << "-" << std::endl;
+	ss << std::left << std::setfill(' ') << std::setw(width) << "State" 
+		 << std::setw(width) << "Expected state" 
+		 << std::setw(width) << "Pass" << std::endl;
+	ss << std::setfill('-') << std::setw(width * 3) << "-" << std::endl;
+
+	bool allStatesCorrect = true;
+	for (auto const& [state, expectedState] : stateResult) {
+		auto pass = (state == expectedState) ? "OK" : "NOT OK";
+		if (pass == "NOT OK") {
+			allStatesCorrect = false;
+		}
+		ss << std::left << std::setfill(' ') << std::setw(width) << state 
+			 << std::setw(width) << expectedState 
+			 << std::setw(width) << pass << std::endl;
+	}
+	ss << std::setfill('-') << std::setw(width * 3) << "-" << std::endl;
+	ss << std::setfill(' ');
+
+	ss << "\nState change result: ";
+	ss << ((allStatesCorrect) ? "OK" : "NOT OK");
+	ss << "\nTrajectory following result: ";
+	ss << ((followedTrajectory) ? "OK" : "NOT OK");
+	RCLCPP_INFO(get_logger(), ss.str().c_str());
+}
+
+
+/**
+ * @brief This is a placeholder callback function. It is needed in order to create a topic subscriber.
+ * 
+ * @param msg Message.
+ */
+void ScenarioExecution::placeholderCallback(const atos_interfaces::msg::Monitor::SharedPtr msg) {}
```

Comment:
Why placeholderCallback and not monrCallback?

Reply:
This is just a function needed for `monitorSub` in `rclcpp::wait_for_message`, this function doesn't do anything but is needed anyway. I named it `placeholderCallback` so it's clear it's just a placeholder.